File: js/tscripts/Ps.js
if (loglvl) print("--> Ps.js");

var RPMPROB_BADARCH	= 0;    /*!< (unused) package ... is for a different architecture */
var RPMPROB_BADOS	= 1;      /*!< (unused) package ... is for a different operating system */
var RPMPROB_PKG_INSTALLED = 2; /*!< package ... is already installed */
var RPMPROB_BADRELOCATE	= 3; /*!<path ... is not relocatable for package ... */
var RPMPROB_REQUIRES	= 4;   /*!< package ... has unsatisfied Requires: ... */
var RPMPROB_CONFLICT	= 5;   /*!< package ... has unsatisfied Conflicts: ... */
var RPMPROB_NEW_FILE_CONFLICT = 6; /*!< file ... conflicts between attemped installs of ... */
var RPMPROB_FILE_CONFLICT = 7; /*!<file ... from install of ... conflicts with file from package ... */
var RPMPROB_OLDPACKAGE	= 8; /*!< package ... (which is newer than ...) is already installed */
var RPMPROB_DISKSPACE	= 9;  /*!< installing package ... needs ... on the ... filesystem */
var RPMPROB_DISKNODES	= 10;  /*!< installing package ... needs ... on the ... filesystem */
var RPMPROB_RDONLY	= 11;     /*!< installing package ... on ... rdonly filesystem */
var RPMPROB_BADPRETRANS	= 12; /*!< (unimplemented) */
var RPMPROB_BADPLATFORM	= 13; /*!< package ... is for a different platform */
var RPMPROB_NOREPACKAGE	= 14; /*!< re-packaged package ... is missing */

var ps = new Ps();
ack("typeof ps;", "object");
ack("ps instanceof Ps;", true);
ack("ps.debug = 1;", 1);
ack("ps.debug = 0;", 0);

var PKG = "PKG";
var ALT = "ALT";
var KEY = "/path/pkg";
var dn = "/root/";
var bn = "file";
var ui = 1234;
var R = "R ALT";
var C = "C ALT";
var A = "ARCH/";
var O = "OS";

ack("ps.length", 0);
// ack("ps.push(PKG, ALT', KEY, RPMPROB_BADARCH, dn, bn, ui)", undefined);
// ack("ps.push(PKG, ALT', KEY, RPMPROB_BADOS, dn, bn, ui)", undefined);
ack("ps.push(PKG, ALT, KEY, RPMPROB_PKG_INSTALLED, dn, bn, ui)", undefined);
ack("ps.push(PKG, ALT, KEY, RPMPROB_BADRELOCATE, dn, bn, ui)", undefined);
ack("ps.push(PKG,   R, KEY, RPMPROB_REQUIRES, dn, bn, ui)", undefined);
ack("ps.push(PKG,   C, KEY, RPMPROB_CONFLICT, dn, bn, ui)", undefined);
ack("ps.push(PKG, ALT, KEY, RPMPROB_NEW_FILE_CONFLICT, dn, bn, ui)", undefined);
ack("ps.push(PKG, ALT, KEY, RPMPROB_FILE_CONFLICT, dn, bn, ui)", undefined);
ack("ps.push(PKG, ALT, KEY, RPMPROB_OLDPACKAGE, dn, bn, ui)", undefined);
ack("ps.push(PKG, ALT, KEY, RPMPROB_DISKSPACE, dn, bn, ui)", undefined);
ack("ps.push(PKG, ALT, KEY, RPMPROB_DISKNODES, dn, bn, ui)", undefined);
ack("ps.push(PKG, ALT, KEY, RPMPROB_RDONLY, dn, bn, ui)", undefined);
// ack("ps.push(PKG, ALT, KEY, RPMPROB_BADPRETRANS, dn, bn, ui)", undefined);
ack("ps.push(PKG, ALT, KEY, RPMPROB_BADPLATFORM, A, O, ui)", undefined);
ack("ps.push(PKG, ALT, KEY, RPMPROB_NOREPACKAGE, dn, bn, ui)", undefined);
ack("ps.length", 12);

ack("ps.print()", undefined);

if (loglvl) print("<-- Ps.js");

File: ruby/tscripts/Ps.rb
if ($loglvl)
  print("--> Ps.rb\n");
end

$RPMPROB_BADARCH     = 0;
$RPMPROB_BADOS       = 1;
$RPMPROB_PKG_INSTALLED = 2;
$RPMPROB_BADRELOCATE = 3;
$RPMPROB_REQUIRES    = 4;
$RPMPROB_CONFLICT    = 5;
$RPMPROB_NEW_FILE_CONFLICT = 6;
$RPMPROB_FILE_CONFLICT = 7;
$RPMPROB_OLDPACKAGE  = 8;
$RPMPROB_DISKSPACE   = 9;
$RPMPROB_DISKNODES   = 10;
$RPMPROB_RDONLY      = 11;
$RPMPROB_BADPRETRANS = 12;
$RPMPROB_BADPLATFORM = 13;
$RPMPROB_NOREPACKAGE = 14;

$ps = Ps.new
ack("$ps.instance_of?(Ps)", true)
ack("$ps.kind_of?(Ps)", true)
ack("$ps.class.to_s", "Ps")
ack("$ps.debug = 1;", 1);
ack("$ps.debug = 0;", 0);

$PKG = "PKG";
$ALT = "ALT";
$KEY = "/path/pkg";
$dn = "/root/";
$bn = "file";
$ui = 1234;
$R = "R ALT";
$C = "C ALT";
$A = "ARCH/";
$O = "OS";

ack("$ps.length", 0);
# ack("$ps.push($PKG, $ALT, $KEY, $RPMPROB_BADARCH, $dn, $bn, $ui)", nil);
# ack("$ps.push($PKG, $ALT, $KEY, $RPMPROB_BADOS, $dn, $bn, $ui)", nil);
ack("$ps.push($PKG, $ALT, $KEY, $RPMPROB_PKG_INSTALLED, $dn, $bn, $ui)", nil);
ack("$ps.push($PKG, $ALT, $KEY, $RPMPROB_BADRELOCATE, $dn, $bn, $ui)", nil);
ack("$ps.push($PKG,   $R, $KEY, $RPMPROB_REQUIRES, $dn, $bn, $ui)", nil);
ack("$ps.push($PKG,   $C, $KEY, $RPMPROB_CONFLICT, $dn, $bn, $ui)", nil);
ack("$ps.push($PKG, $ALT, $KEY, $RPMPROB_NEW_FILE_CONFLICT, $dn, $bn, $ui)", nil);
ack("$ps.push($PKG, $ALT, $KEY, $RPMPROB_FILE_CONFLICT, $dn, $bn, $ui)", nil);
ack("$ps.push($PKG, $ALT, $KEY, $RPMPROB_OLDPACKAGE, $dn, $bn, $ui)", nil);
ack("$ps.push($PKG, $ALT, $KEY, $RPMPROB_DISKSPACE, $dn, $bn, $ui)", nil);
ack("$ps.push($PKG, $ALT, $KEY, $RPMPROB_DISKNODES, $dn, $bn, $ui)", nil);
ack("$ps.push($PKG, $ALT, $KEY, $RPMPROB_RDONLY, $dn, $bn, $ui)", nil);
# ack("$ps.push($PKG, $ALT, $KEY, $RPMPROB_BADPRETRANS, $dn, $bn, $ui)", nil);
ack("$ps.push($PKG, $ALT, $KEY, $RPMPROB_BADPLATFORM, $A, $O, $ui)", nil);
ack("$ps.push($PKG, $ALT, $KEY, $RPMPROB_NOREPACKAGE, $dn, $bn, $ui)", nil);
ack("$ps.length", 12);

if ($loglvl)
  print("<-- Ps.rb\n");
end
